### PR TITLE
Install package in dockerfile

### PR DESCRIPTION
**Pull request by matt-peters**
Installing allennlp inside the `Dockerfile` allows other projects that depend on allennlp to use the docker container as a base image.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -52,6 +52,9 @@ COPY .pylintrc .pylintrc
 COPY tutorials/ tutorials/
 COPY training_config training_config/
 COPY setup.py setup.py
+COPY README.md README.md
+
+RUN python setup.py install
```
Comment (schmmd) [inline Dockerfile]: I would use `pip install --editable .` instead.

**Comment by schmmd** [comment]
@matt-peters this seems like a good change but I'm curious--what's your specific motivation for it?  For each commit to master we publish a simpler Docker image with that version of AllenNLP pip installed.  See https://github.com/allenai/allennlp/blob/master/Dockerfile.pip.

**Comment by matt-peters** [comment]
This allows someone to build a base image from a branch that hasn't been merged into master.

**Comment by schmmd** [comment]
@matt-peters  just change `RUN python setup.py install` to `pip install --editable .` and merge it!

**Pull request merged**